x-pack/plugins/code/server/lsp/ts_launcher.ts

```diff
@@ -4,109 +4,58 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { spawn } from 'child_process';
+import { ChildProcess, spawn } from 'child_process';
 import getPort from 'get-port';
 import { resolve } from 'path';
 import { Logger } from '../log';
 import { ServerOptions } from '../server_options';
 import { LoggerFactory } from '../utils/log_factory';
-import { ILanguageServerLauncher } from './language_server_launcher';
 import { LanguageServerProxy } from './proxy';
 import { RequestExpander } from './request_expander';
+import { AbstractLauncher } from './abstract_launcher';
 
-export class TypescriptServerLauncher implements ILanguageServerLauncher {
-  private isRunning: boolean = false;
-  constructor(
+export class TypescriptServerLauncher extends AbstractLauncher {
+  public constructor(
     readonly targetHost: string,
     readonly options: ServerOptions,
     readonly loggerFactory: LoggerFactory
-  ) {}
-
-  public get running(): boolean {
-    return this.isRunning;
+  ) {
+    super('typescript', targetHost, options, loggerFactory);
   }
 
-  public async launch(builtinWorkspace: boolean, maxWorkspace: number, installationPath: string) {
-    let port = 2089;
-
+  async getPort() {
     if (!this.options.lsp.detach) {
-      port = await getPort();
+      return await getPort();
     }
-    const log: Logger = this.loggerFactory.getLogger(['code', `ts@${this.targetHost}:${port}`]);
-    const proxy = new LanguageServerProxy(port, this.targetHost, log, this.options.lsp);
+    return 2089;
```

Comment:
nit: maybe a const at the beginning of this class instead of a hard-wired value.